hazelcast/src/main/java/com/hazelcast/sql/impl/expression/datetime/ExtractField.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.datetime;
+
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoField;
+import java.time.temporal.IsoFields;
+import java.time.temporal.WeekFields;
+
+public enum ExtractField {
+    CENTURY {
+        @Override
+        public double extract(OffsetDateTime time) {
+            int year = time.getYear();
+            int century = (year / YEARS_IN_CENTURY) + FIRST_CENTURY;
+            return century;
+        }
+    },
+    DAY {
+        @Override
+        public double extract(OffsetDateTime time) {
+            return time.get(ChronoField.DAY_OF_MONTH);
```

Comment:
This is probably a bit faster:
```suggestion
            return time.getDayOfMonth();
```